### PR TITLE
chore(stdlib): Convert while loops to for loops in Array (#1009)

### PR DESCRIPTION
**Pull request by timgestson**
Replace while loops where it makes sense with for loops satisfying #1009

### CODE REVIEW

File: stdlib/array.gr [hunk 1]
```diff
@@ -243,15 +243,10 @@ export let copy = (array) => {
  */
 export let cycle = (fn, n, array) => {
   let length = length(array)
-  let mut iteration = 0
-  let mut count = 0
-  while (iteration < n) {
-    while (count < length) {
-      fn(array[count]): Void
-      count = incr(count)
+  for (let mut iteration = 0; iteration < n; iteration = incr(iteration)) {
```
[the remark this answers is not in the document]
Comment (timgestson) [reply stdlib/array.gr]: no problem.  done

**Review by marcusroberts — APPROVED** [review]
Great work, thank you!

File: stdlib/array.gr [hunk 2]
```diff
@@ -681,14 +670,10 @@ export let count = (fn, array) => {
  */
 export let counti = (fn, array) => {
   let length = length(array)
-  let mut position = 0
   let mut count = 0
-  while (position < length) {
-    if(fn(array[position], position)) {
+  for (let mut position = 0; position < length; position = incr(position)) {
```
Comment (peblair) [inline stdlib/array.gr:674]:
```suggestion
  for (let mut position = 0; position < length; position += 1) {
```

**Review by peblair — COMMENTED** [review]
Thank you for this! I caught one lingering `incr`, but this looks good to merge otherwise.

File: stdlib/array.gr (hunk 2 above)
Comment (timgestson) [reply stdlib/array.gr]: Replaced all remaining `incr` calls in the file

**Review by ospencer — APPROVED** [review]
Beautiful, thanks @timgestson! 😄

**Pull request merged**